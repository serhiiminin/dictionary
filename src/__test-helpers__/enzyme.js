import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
