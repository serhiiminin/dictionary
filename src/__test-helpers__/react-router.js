const mockRouterProps = {
  match: {
    isExact: true,
    params: {},
    path: "",
    url: ""
  },
  location: {
    hash: "",
    key: "",
    pathname: "",
    search: "",
    state: {}
  },
  history: {
    length: 2,
    action: "POP",
    location: location,
    push: () => {},
    replace: () => {},
    go: () => {},
    goBack: () => {},
    goForward: () => {},
    block: () => {},
    createHref: () => {},
    listen: () => {}
  },
  staticContext: {}
};

export default mockRouterProps;
