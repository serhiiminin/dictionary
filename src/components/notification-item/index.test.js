import React from "react";
import { shallow } from "enzyme";
import { mountWithTheme } from "../../__test-helpers__/enzyme";
import muiTheme from "../../root/mui-theme";
import NotificationItem, {
  NotificationItemWrapper,
  TopLine,
  WrapperCloseButton
} from ".";

describe("Notification item styled components", () => {
  test("NotificationItemWrapper", () => {
    const wrapper = mountWithTheme(
      <NotificationItemWrapper>Anything</NotificationItemWrapper>,
      muiTheme
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("TopLine", () => {
    const wrapper = mountWithTheme(<TopLine>Anything</TopLine>, muiTheme);

    expect(wrapper).toMatchSnapshot();
  });

  test("WrapperCloseButton", () => {
    const wrapper = mountWithTheme(
      <WrapperCloseButton>Anything</WrapperCloseButton>,
      muiTheme
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("Notification item", () => {
  test("render", () => {
    const wrapper = shallow(
      <NotificationItem onClick={() => {}} status="exited" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
