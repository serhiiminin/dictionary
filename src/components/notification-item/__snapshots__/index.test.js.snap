// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification item render 1`] = `
<WithTheme(Grow)
  in={true}
  timeout={400}
>
  <li>
    <div>
      <pure(Error) />
      <div>
        <Jss(CloseButton)
          onClick={[Function]}
        />
      </div>
    </div>
    <div />
  </li>
</WithTheme(Grow)>
`;
